### PR TITLE
Online pagination integration fixes

### PR DESCRIPTION
**Pull request by osher-sade**

### CODE REVIEW

File: Jenkinsfile [hunk 1]
```diff
@@ -59,6 +59,10 @@ node {
                    echo "release branch: ${env.BRANCH_NAME}"
                    sh "npm run publish-beta"
                }
+               if (env.BRANCH_NAME == "online-pagination-integration-fixes") {
```
Comment (ronkatz96) [inline Jenkinsfile:62]: why?

File: packages/polaris-core/src/plugins/snapshot/snapshot-listener.ts [hunk 2]
```diff
@@ -116,7 +118,7 @@ export class SnapshotListener implements GraphQLRequestListener<PolarisGraphQLCo
     ): Promise<GraphQLResponse | null> | GraphQLResponse | null {
         const { context } = requestContext;
 
-        if (context.snapshotContext) {
+        if (context.returnedExtensions.totalCount !== undefined) {
```
Comment (ronkatz96) [inline packages/polaris-core/src/plugins/snapshot/snapshot-listener.ts:121]: change to != null

File: packages/polaris-core/src/server/routes/snapshot-routes.ts [hunk 3]
```diff
@@ -87,6 +90,24 @@ export const createSnapshotRoutes = (polarisServerConfig: PolarisServerConfig):
     return router;
 };
 
+function getConnection(
```
Comment (ronkatz96) [inline packages/polaris-core/src/server/routes/snapshot-routes.ts:93]: add return type

File: packages/polaris-test/tests/date-filter.test.ts [hunk 4]
```diff
@@ -11,7 +11,7 @@ function sleep(ms: number) {
     return new Promise((resolve) => setTimeout(resolve, ms));
 }
 
-describe('date filter tests', () => {
+describe.skip('date filter tests', () => {
```
Comment (ronkatz96) [inline packages/polaris-test/tests/date-filter.test.ts:14]: fix or add todo

File: packages/polaris-common/src/common-methods.ts [hunk 5]
```diff
@@ -1,6 +1,10 @@
-export const runAndMeasureTime = async (runnable: () => {}) => {
+export async function runAndMeasureTime(runnable: () => {}) {
```
Comment (itaykl) [inline packages/polaris-common/src/common-methods.ts:1]: maybe put this file inside utils directory, and modify the file name to utils-methods

File: packages/polaris-middlewares/src/transactional-requests-plugin/transactional-requests-plugin.ts [hunk 6]
```diff
@@ -28,10 +27,7 @@ export class TransactionalMutationsPlugin implements ApolloServerPlugin<PolarisG
     public requestDidStart(
         requestContext: GraphQLRequestContext<PolarisGraphQLContext>,
     ): GraphQLRequestListener<PolarisGraphQLContext> | void {
-        if (
-            isMutation(requestContext.request.query) &&
-            this.connectionManager?.connections?.length
-        ) {
+        if (this.connectionManager?.connections?.length) {
```
Comment (itaykl) [inline packages/polaris-middlewares/src/transactional-requests-plugin/transactional-requests-plugin.ts:30]: why did you delete the isMutation condition?

File: packages/polaris-core/src/middlewares/snapshot-middleware.ts [hunk 7]
```diff
@@ -26,10 +26,17 @@ export class SnapshotMiddleware {
...
-                    : await result.getData(0, await result.totalCount());
+            if (context.requestHeaders.snapRequest || this.config.snapshotConfig.autoSnapshot) {
+                currentPage = await this.calculateCurrentPageInSnapshotProcess(context, result);
+            } else {
+                const totalCount =
+                    context.returnedExtensions.totalCount ?? (await result.totalCount());
+                if (totalCount < this.config.maxPageSize) {
+                    currentPage = await result.getData(0, totalCount);
+                } else {
+                    throw new Error('snap-request header is missing');
```
Comment (itaykl) [inline packages/polaris-core/src/middlewares/snapshot-middleware.ts:37]: i dont understand how this else reflects the error thrown inside

File: packages/polaris-core/src/middlewares/snapshot-middleware.ts [hunk 8]
```diff
@@ -44,7 +51,7 @@ export class SnapshotMiddleware {
         context: PolarisGraphQLContext,
         result: any,
     ) {
-        if (context.snapshotContext == null || context.snapshotContext.startIndex === 0) {
+        if (context.snapshotContext == null || !context.snapshotContext.startIndex) {
```
Comment (itaykl) [inline packages/polaris-core/src/middlewares/snapshot-middleware.ts:54]: i think that for sake of readability undo that and leave it === 0, unless its for checking for nulls/undefind etc

File: packages/polaris-core/src/plugins/snapshot/snapshot-listener.ts [hunk 9]
```diff
@@ -78,17 +77,20 @@ export class SnapshotListener implements GraphQLRequestListener<PolarisGraphQLCo
 
         if (this.isSnapshotRequest(context, requestContext.request.query)) {
             return (async (): Promise<void> => {
+                const connection = getConnectionForReality(
```
Comment (itaykl) [inline packages/polaris-core/src/plugins/snapshot/snapshot-listener.ts:80]: add support for connectionless logic

File: packages/polaris-core/src/server/routes/snapshot-routes.ts [hunk 10]
```diff
@@ -19,20 +19,23 @@ export async function snapshotPageRoute(
...
-    const result = await getSnapshotPageById(id, realityId, polarisServerConfig, connection as any);
-    if (!result) {
-        res.send({ message: `Snapshot page with id ${id} not found` });
-    } else {
-        const responseToSend =
-            result!.status !== SnapshotStatus.DONE
-                ? { status: result!.status, id: result!.id }
-                : result!.getData();
-        res.send(responseToSend);
+    const connection = getConnection(realityId, polarisServerConfig, res);
```
Comment (itaykl) [inline packages/polaris-core/src/server/routes/snapshot-routes.ts:22]: check for support to connectionless logic

File: packages/polaris-core/src/server/routes/snapshot-routes.ts [hunk 11]
```diff
@@ -44,32 +47,32 @@ export async function snapshotMetadataRoute(
...
-        clean(formattedResult);
-        if (result.errors) {
-            formattedResult.errors = JSON.parse(result.errors.toString());
-        }
-        if (result.warnings) {
-            formattedResult.warnings = JSON.parse(result.warnings.toString());
-        }
-        if (result.irrelevantEntities) {
-            formattedResult.irrelevantEntities = JSON.parse(result.irrelevantEntities.toString());
+    const connection = getConnection(realityId, polarisServerConfig, res);
```
Comment (itaykl) [inline packages/polaris-core/src/server/routes/snapshot-routes.ts:50]: check for support to connectionless logic

File: packages/polaris-core/src/server/routes/snapshot-routes.ts [hunk 12]
```diff
@@ -87,6 +90,24 @@ export const createSnapshotRoutes = (polarisServerConfig: PolarisServerConfig):
...
 };
 
+function getConnection(
+    realityId: number,
+    polarisServerConfig: PolarisServerConfig,
+    res: express.Response,
+) {
+    let connection;
+    try {
+        connection = getConnectionForReality(
```
Comment (itaykl) [inline packages/polaris-core/src/server/routes/snapshot-routes.ts:100]: add support to connectionless

File: packages/polaris-typeorm/src/handlers/data-version-handler.ts [hunk 13]
```diff
@@ -1,72 +1,60 @@
...
+        connection: PolarisConnection,
+    ): Promise<{ id: number; value: number } | undefined> {
         try {
-            if (!runner.isTransactionActive) {
-                await runner.startTransaction();
-            }
-            result = await runner.manager
-                .getRepository(DataVersion)
+            return manager
                 .createQueryBuilder()
```
Comment (itaykl) [inline packages/polaris-typeorm/src/handlers/data-version-handler.ts:49]: check if this logic dont fails the connectionless logic

File: packages/polaris-typeorm/src/handlers/data-version-handler.ts [hunk 14]
```diff
@@ -1,72 +1,60 @@
...
 import { EntityMetadata } from 'typeorm';
 import { RelationMetadata } from 'typeorm/metadata/RelationMetadata';
-import { DataVersion, PolarisConnection, QueryRunner, SelectQueryBuilder } from '..';
+import { DataVersion, PolarisConnection, PolarisEntityManager, SelectQueryBuilder } from '..';
 import { isDescendentOfCommonModel } from '../utils/descendent-of-common-model';
 
 export class DataVersionHandler {
     public async updateDataVersion<Entity>(
-        context: PolarisGraphQLContext,
         connection: PolarisConnection,
```
Comment (itaykl) [inline packages/polaris-typeorm/src/handlers/data-version-handler.ts:9]: you are using connection only for the purpose of logging, if so, send the logger object and use it instead

File: packages/polaris-typeorm/src/handlers/data-version-handler.ts [hunk 15]
```diff
@@ -1,72 +1,60 @@
...
-        runner: QueryRunner,
+        manager: PolarisEntityManager,
     ) {
         const extensions: PolarisExtensions =
-            (context && context.returnedExtensions) || ({} as PolarisExtensions);
+            (manager.context && manager.context.returnedExtensions) || ({} as PolarisExtensions);
         connection.logger.log('log', 'Started data version job when inserting/updating entity');
-        const result = await this.getDataVersionForMutation(runner, connection);
+        const result = await this.selectDataVersionForUpdate(manager, connection);
```
Comment (itaykl) [inline packages/polaris-typeorm/src/handlers/data-version-handler.ts:15]: you are using connection only for the purpose of logging, if so, send the logger object and use it instead.

File: packages/polaris-typeorm/src/handlers/data-version-handler.ts [hunk 16]
```diff
@@ -168,10 +156,11 @@ function joinDataVersionRelations(
     entityMetadata: EntityMetadata,
     names: string[],
 ): SelectQueryBuilder<any> {
+    let entityIdAlias = entityMetadata.name;
+    if (findSorted) {
+        qb.select(entityIdAlias + '.id', entityIdAlias);
```
Comment (itaykl) [inline packages/polaris-typeorm/src/handlers/data-version-handler.ts:161]: string.format

File: packages/polaris-typeorm/src/handlers/data-version-handler.ts [hunk 17]
```diff
@@ -180,6 +169,16 @@ function joinDataVersionRelations(
             !findSorted,
         );
     }
+    if (qb.expressionMap.selects.length <= 1 && findSorted) {
+        entityIdAlias = 'entityId';
+        qb.select(entityName + '.id', entityIdAlias);
```
Comment (itaykl) [inline packages/polaris-typeorm/src/handlers/data-version-handler.ts:174]: string.format

File: packages/polaris-typeorm/src/handlers/data-version-handler.ts [hunk 18]
```diff
@@ -180,6 +169,16 @@ function joinDataVersionRelations(
...
+    }
+    if (findSorted) {
+        qb.addSelect(
+            entityIdAlias + '.dataVersion',
```
Comment (itaykl) [inline packages/polaris-typeorm/src/handlers/data-version-handler.ts:178]: string.format

File: packages/polaris-typeorm/src/handlers/find-handler.ts [hunk 19]
```diff
@@ -14,27 +9,42 @@ const realityIdCriteria = (includeLinkedOper: boolean, headers: PolarisRequestHe
...
         }
         if (polarisCriteria.where.realityId === undefined) {
             polarisCriteria.where.realityId = realityIdCriteria(includeLinkedOper, headers);
         }
         return polarisCriteria;
     }
+
+    private deleteFindConditionIfRedundant(polarisCriteria: any) {
+        if (
+            polarisCriteria.where.deleted?._type === 'in' &&
```
Comment (itaykl) [inline packages/polaris-typeorm/src/handlers/find-handler.ts:41]: may god help us (;

**Review by chenshoo — APPROVED** [review]
